mcuboot: Add S2RAM early hooks (#28040)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

> * Fixed an issue where the MCUboot was unable to recover after the suspend to RAM if the manifest-based updates were enabled for nRF54H20 devices.

Reviewer: ```suggestion
* Fixed an issue where MCUboot could not recover after resuming from suspend-to-RAM when manifest-based updates were enabled on nRF54H20 devices.
```
